MonoGame.Framework/Platform/Native/Platform.Interop.cs

```diff
@@ -0,0 +1,299 @@
+
+    [LibraryImport(MonoGameNativeDLL, EntryPoint = "MGP_Platform_Exit", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial void Platform_Exit(MGP_Platform* platform);
+
+    #endregion
+
+    #region Window
+
+    [LibraryImport(MonoGameNativeDLL, EntryPoint = "MGP_Window_Create", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial MGP_Window* Window_Create(
+        MGP_Platform* platform,
+        int width,
+        int height,
+        string title);
+
+    [LibraryImport(MonoGameNativeDLL, EntryPoint = "MGP_Window_Destroy", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial void Window_Destroy(MGP_Window* window);
+
+    [LibraryImport(MonoGameNativeDLL, EntryPoint = "MGP_Window_GetNativeHandle", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial nint Window_GetNativeHandle(MGP_Window* window);
+
+    [LibraryImport(MonoGameNativeDLL, EntryPoint = "MGP_Window_GetAllowUserResizing", StringMarshalling = StringMarshalling.Utf8)]
+    [return: MarshalAs(UnmanagedType.U1)]
+    public static partial bool Window_GetAllowUserResizing(MGP_Window* window);
+
+    [LibraryImport(MonoGameNativeDLL, EntryPoint = "MGP_Window_SetAllowUserResizing", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial void Window_SetAllowUserResizing(MGP_Window* window, [MarshalAs(UnmanagedType.U1)] bool allow);
+
+    [LibraryImport(MonoGameNativeDLL, EntryPoint = "MGP_Window_GetIsBoderless", StringMarshalling = StringMarshalling.Utf8)]
+    [return: MarshalAs(UnmanagedType.U1)]
+    public static partial bool Window_GetIsBorderless(MGP_Window* window);
+
+    [LibraryImport(MonoGameNativeDLL, EntryPoint = "MGP_Window_SetIsBoderless", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial void Window_SetIsBorderless(MGP_Window* window, [MarshalAs(UnmanagedType.U1)] bool borderless);
+
+    [LibraryImport(MonoGameNativeDLL, EntryPoint = "MGP_Window_SetTitle", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial void Window_SetTitle(MGP_Window* window, string title);
+
+    [LibraryImport(MonoGameNativeDLL, EntryPoint = "MGP_Window_Show", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial void Window_Show(MGP_Window* window, [MarshalAs(UnmanagedType.U1)] bool show);
+
+    [LibraryImport(MonoGameNativeDLL, EntryPoint = "MGP_Window_GetPosition", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial void Window_GetPosition(MGP_Window* window, out int x, out int y);
+
+    [LibraryImport(MonoGameNativeDLL, EntryPoint = "MGP_Window_SetPosition", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial void Window_SetPosition(MGP_Window* window, int x, int y);
+
+    #endregion
+
+    #region Mouse
+
+    [LibraryImport(MonoGameNativeDLL, EntryPoint = "MGP_Mouse_SetVisible", StringMarshalling = StringMarshalling.Utf8)]
```

Comment:
Should i split out input stuff into its own interop static class?  Or place it all in `MGP` platform class?

Reply:
It would make sense to have them in Mouse.Native and GamePad.Native.

Reply:
How the type generator works is it finds static classes in the `MonoGame.Interop.` namespace that have pinvokes.

So just making the `LibraryImport` members of the `Mouse` static class isn't enough to get them exported unless we change how these are found.  An attribute on the class maybe?

Thoughts @harry-cpp ?

Do we want/need more organization to the pinvokes than we have now?